examples/step-57/step-57.cc

```diff
@@ -0,0 +1,949 @@
+#include <deal.II/base/utilities.h>
+
+#include <deal.II/lac/block_vector.h>
+#include <deal.II/lac/full_matrix.h>
+#include <deal.II/lac/block_sparse_matrix.h>
+#include <deal.II/lac/dynamic_sparsity_pattern.h>
+#include <deal.II/lac/solver_cg.h>
+#include <deal.II/lac/solver_gmres.h>
+#include <deal.II/lac/precondition.h>
+#include <deal.II/lac/constraint_matrix.h>
+#include <deal.II/lac/sparse_direct.h>
+
+#include <deal.II/grid/tria.h>
+#include <deal.II/grid/grid_generator.h>
+#include <deal.II/grid/tria_accessor.h>
+#include <deal.II/grid/tria_iterator.h>
+#include <deal.II/grid/manifold_lib.h>
+#include <deal.II/grid/grid_refinement.h>
+#include <deal.II/grid/tria_boundary_lib.h>
+#include <deal.II/grid/grid_tools.h>
+
+#include <deal.II/dofs/dof_handler.h>
+#include <deal.II/dofs/dof_renumbering.h>
+#include <deal.II/dofs/dof_accessor.h>
+#include <deal.II/dofs/dof_tools.h>
+
+#include <deal.II/fe/fe_q.h>
+#include <deal.II/fe/fe_system.h>
+#include <deal.II/fe/fe_values.h>
+
+#include <deal.II/numerics/vector_tools.h>
+#include <deal.II/numerics/matrix_tools.h>
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/numerics/error_estimator.h>
+
+// To transfer solutions between meshes, this file is included.
+#include <deal.II/numerics/solution_transfer.h>
+
+// This file includes UMFPACK: the direct solver.
+#include <deal.II/lac/sparse_direct.h>
+
+// And the one for ILU preconditioner
+#include <deal.II/lac/sparse_ilu.h>
+
+// C++
+#include <fstream>
+#include <iostream>
+#include <sstream>
+
+// As in all programs, the namespace dealii is included
+namespace Step57
+{
+  using namespace dealii;
+
+  // @sect3{The <code>StokesProblem</code> class template}
+
+  // As explained in introduction, what we obtain at each step is the Newton's update term, so
+  // we define two variables: the present solution and the update term. By Newton's iteration, the new solution can be written
+  // as $x_{new} = x_{old} + x_{update}$. Additionally, the evaluation point is for temporarily holding the updated solution
+  // in line search. A sparse matrix for the mass matrix of pressure is created for the operator of
+  // a block Schur complement preconditioner. We use one ConstraintMatrix for Dirichlet boundary conditions at
+  // the initial step and a zero ConstraintMatrix for the Newton's update term. $\gamma$ is the weight of augmented lagrangian
+  // term.
+
+  template <int dim>
+  class Navier_Stokes_Newton
+  {
+  public:
+    Navier_Stokes_Newton(const unsigned int degree);
+    ~Navier_Stokes_Newton();
+    void run();
+
+  private:
+    void setup_system(bool setup_dof, bool initialize_system);
+    void assemble_NavierStokes_system(bool initial_step,
+                                      bool assemble_matrix,
+                                      bool assemble_rhs);
+    void solve(bool initial_step);
+    void refine_mesh();
+    void process_solution();
+    void output_results (const unsigned int refinement_cycle) const;
```

Comment:
make ``process_solution`` part of ``output_results`` and create ``line_xy.txt`` in every cycle.